solr/core/src/java/org/apache/solr/core/QuerySenderListener.java

```diff
@@ -46,7 +46,8 @@ public void newSearcher(SolrIndexSearcher newSearcher, SolrIndexSearcher current
     final SolrIndexSearcher searcher = newSearcher;
     log.debug("QuerySenderListener sending requests to {}", newSearcher);
     @SuppressWarnings("unchecked")
-    List<NamedList<Object>> allLists = (List<NamedList<Object>>) getArgs().get("queries");
+    ArrayList<NamedList<Object>> allLists =
```

Comment:
OK, I've switched `allLists` back to being a `List` - though `queries2list()` will always create an `ArrayList` because `List` can't be instantiated. I don't have a strong preference either way - happy to take your guidance!